source/Files.cpp

```diff
@@ -561,6 +549,21 @@ void Files::Write(FILE *file, const string &data)
 
 
 
+void Files::CreateDirectory(const std::string &path)
```

Comment:
windows `#defines CreateDirectory` to its own garbage. you will want a different name here.

Reply:
oh right I forgot about that, thanks

Reply:
If you want to use MakeDir, then I won't have to rewrite this: https://github.com/thewierdnut/endless-mobile/blob/android/source/Files.cpp#L689

Reply:
Does that function create any directory that doesn't exist in the given path? Why/where do you need this?

Reply:
I'm not saying steal that whole function, just proposing an alternative name.

Reply:
And yeah... the implementation I wrote recursively creates all the directories (think `mkdir -p`) in the given path. That was needed for zip processing though, which you guys don't have.

Reply:
oh I see, idk I find CreateFolder clearer

Reply:
Meh. I moved all the zip processing into the java code. I can probably delete my code.

Reply:
on an unrelated note, did you see this https://github.com/libsdl-org/SDL/issues/8129

Reply:
Now that std::filesystem is a thing, it seems pointless. I wish we had this 10 years ago.

Reply:
Can std::filesystem be used on Android? if so that's dope

Reply:
hm... It looks like it is there, though I haven't tried it yet:
```
android-ndk-r20b$ find . -iname filesystem
./toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/filesystem
./toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/experimental/filesystem
./sources/cxx-stl/llvm-libc++/include/filesystem
./sources/cxx-stl/llvm-libc++/include/experimental/filesystem
./sources/cxx-stl/llvm-libc++/test/std/experimental/filesystem
./sources/cxx-stl/llvm-libc++/test/libcxx/experimental/filesystem
```